### ISSUE TITLE
Make ICG compatible with clang/llvm 15

### ISSUE THREAD

**Issue by alexlin0** [nasa/trick#1449]

**Comment by sochsner1** [comment]
Exactly what I came here to say. clang 15 proliferated in to my packages mirror and now `make` breaks on HeaderSearchDirs.cpp.
This is the clang commit that broke it: https://github.com/llvm/llvm-project/commit/7631c366c8589dda488cb7ff1df26cc134002208

```/usr/bin/g++ -g -I/usr/include -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -fno-rtti -I/usr/include/udunits2 -std=c++14 -DLIBCLANG_MAJOR=15 -DLIBCLANG_MINOR=0 -DLIBCLANG_PATCHLEVEL=0 -DTRICK_GCC_VERSION=\"8.5.0\" -DLLVM_HOME=\"/usr\" -c HeaderSearchDirs.cpp -o object_Linux_8.5_x86_64/HeaderSearchDirs.o
HeaderSearchDirs.cpp: In member function 'void HeaderSearchDirs::ApplyHeaderSearchOptions()':
HeaderSearchDirs.cpp:187:26: error: 'search_dir_iterator' is not a member of 'clang::HeaderSearch'
     clang::HeaderSearch::search_dir_iterator sdi ;
                          ^~~~~~~~~~~~~~~~~~~
```

Starting with the above commit clang::HeaderSearch::search_dir_iterator no longer exists

**Comment by jmpenn** [comment]
Resolved with merge of #1450.
